signing/signer.go

```diff
@@ -160,6 +176,19 @@ func NewEdSigner(opts ...EdSignerOptionFunc) (*EdSigner, error) {
 			return nil, fmt.Errorf("could not generate key pair: %w", err)
 		}
 		cfg.priv = priv
+
+		if cfg.file != "" {
+			if _, err := os.Stat(cfg.file); err == nil {
```

Comment:
It would be better to also check `errors.Is(err, os.ErrNotExist)`. If there's a permission issue, "already exists" error may be misleading